cpp/src/dictionary/add_keys.cu

```diff
@@ -58,25 +59,28 @@ std::unique_ptr<column> add_keys(
   auto combined_keys =
     cudf::detail::concatenate(std::vector<column_view>{old_keys, new_keys}, stream);
   // sort and remove any duplicates from the combined keys
-  // drop_duplicates([a,b,c,d,f,d,b,e]) = [a,b,c,d,e,f]
-  auto table_keys = cudf::detail::drop_duplicates(table_view{{combined_keys->view()}},
-                                                  std::vector<size_type>{0},  // only one key column
-                                                  duplicate_keep_option::KEEP_FIRST,
-                                                  null_equality::EQUAL,
-                                                  null_order::BEFORE,
-                                                  stream,
-                                                  mr)
-                      ->release();
-  std::unique_ptr<column> keys_column(std::move(table_keys.front()));
+  // unordered_drop_duplicates([a,b,c,d,f,d,b,e]) = [a,b,c,d,e,f]
+  auto table_keys =
+    cudf::detail::unordered_drop_duplicates(table_view{{combined_keys->view()}},
+                                            std::vector<size_type>{0},  // only one key column
+                                            null_equality::EQUAL,
+                                            stream,
+                                            mr);
+
+  std::vector<order> column_order{order::ASCENDING};
+  std::vector<null_order> null_precedence{null_order::AFTER};  // should be no nulls here
+  auto sorted_keys =
+    cudf::detail::sort(table_keys->view(), column_order, null_precedence, stream, mr)->release();
```

Comment:
I liked how the `column_order` and `null_precedence` vectors were defined inline in the other call in `concatenate.cu`. No need to name these variables in my opinion, since their enum values are self-explanatory.
https://github.com/rapidsai/cudf/pull/10030/files#diff-e57bb56f60427be2c2bab0e098d9c965996d0516867693372f72982ecb319cfeR222-R224

Reply:
They are defined inline in `concatenate.cu` since it's used only once. Here `column_order` and `null_precedence` will be used later in `cudf::detail::lower_bound` (see [here)](https://github.com/rapidsai/cudf/pull/10030/files#diff-1515dfb9b55da21e427d3ccd4c700064e6b8d5357f865903e09dbd92a3aa64e0R80-R81) as well so I decided to defined those two vectors.

Reply:
Got it! Leave it as it is, then. Sorry for missing that.